scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontendState.java

```diff
@@ -1230,4 +1231,22 @@ public UserData getCurrentUserData() throws NotConnectedException {
         renewUserSession(id, ident);
         return PASchedulerProperties.getPropertiesAsHashMap();
     }
+
+    synchronized TaskStatesPage getTaskPaginated(JobId jobId, int offset, int limit)
+            throws UnknownJobException, NotConnectedException, PermissionException {
+        checkPermissions("getTaskPaginated",
```

Comment:
all getTaskState methods actually use the "getJobState" permission. If a user has the permission to read the state of a given job, then he can read the state of any of its tasks.

Reply:
So you suggest to remove `checkPermissions` here?

Reply:
no, to use checkPermissions("getJobState", ...)